### PR TITLE
netdiscover: init at 0.8.1

### PR DESCRIPTION
**Pull request by vdot0x23**
###### Motivation for this change

Add https://github.com/netdiscover-scanner/netdiscover helping us become hackers #81418

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by RasmusRendal — APPROVED** [review]

File: pkgs/tools/networking/netdiscover/default.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
+{ lib, stdenv, fetchFromGitHub, libpcap, libnet, autoreconfHook }:
+
+stdenv.mkDerivation rec {
+  pname = "netdiscover";
+  version = "0.8.1";
+
+  src = fetchFromGitHub {
+    owner = "netdiscover-scanner";
+    repo = pname;
+    rev = "${version}";
```
Comment (sikmir) [inline pkgs/tools/networking/netdiscover/default.nix:10]:
```suggestion
    rev = version;
```

File: pkgs/tools/networking/netdiscover/default.nix [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  version = "0.8.1";
+
+  src = fetchFromGitHub {
+    owner = "netdiscover-scanner";
+    repo = pname;
+    rev = version;
+    sha256 = "13fp9rfr9vh756m5wck76zbcr0296ir52dahzlqdr52ha9vrswbb";
+  };
+
+  buildInputs = [ libpcap libnet autoreconfHook ];
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/netdiscover/default.nix:14]: Please Split autoreonfHook into nativeBuildInputs.

File: pkgs/tools/networking/netdiscover/default.nix [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+  autoreconfPhase = "./autogen.sh";
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/netdiscover/default.nix:16]: Isn't the hook doing that already?

**Comment by SuperSandro2000** [comment]
Also please squash the commits for the package into one.

File: pkgs/tools/networking/netdiscover/default.nix (hunk 3 above)
Comment (vdot0x23) [reply pkgs/tools/networking/netdiscover/default.nix]: I think the hook runs `autoconf`but that is also what `autogen.sh` does https://github.com/netdiscover-scanner/netdiscover/blob/master/autogen.sh

**Comment by vdot0x23** [comment]
> Also please squash the commits for the package into one.

Should be good now :)

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 145184` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>netdiscover</li>
  </ul>
</details>

**Review by fabaff — APPROVED** [review]
- [x] package path fits guidelines
- [x] package name fits guidelines
- [x] package version fits guidelines
- [x] `meta.description` is set and fits guidelines
- [x] `meta.license` fits upstream license
- [x] `meta.maintainers` is set
- [x] package builds (see https://github.com/NixOS/nixpkgs/pull/145184#issuecomment-974874300)
- [ ] all depending packages build
- [x] tested on x86_64

```bash
...ew/pr-145184]$ sudo ./results/netdiscover/bin/netdiscover -h
Netdiscover 0.8.1 [Active/passive ARP reconnaissance tool]
Written by: Jaime Penalba <jpenalbae@gmail.com>

Usage: ./results/netdiscover/bin/netdiscover [-i device] [-r range | -l file | -p] [-m file] [-F filter] [-s time] [-c count] [-n node] [-dfPLNS]
  -i device: your network device
[...]
```

Llooks good to me..

**Pull request merged**